include/aws/s3/s3.h

```diff
@@ -50,6 +50,7 @@ enum aws_s3_errors {
     AWS_ERROR_S3_RECV_FILE_NOT_FOUND,
     AWS_ERROR_S3_REQUEST_TIMEOUT,
     AWS_ERROR_S3_TOKEN_EXPIRED,
+    AWS_ERROR_S3_BUFFER_ALLOCATION_FAILED,
```

Comment:
We should merge #518 before this PR so that removing the error code is not a breaking change.